plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiConfig.java

```diff
@@ -0,0 +1,234 @@
+    public int getPartitionScannerParallelism()
+    {
+        return this.partitionScannerParallelism;
+    }
+
+    @Config("hudi.split-generator-parallelism")
+    @ConfigDescription("Number of threads to use for split generation")
+    public HudiConfig setSplitGeneratorParallelism(int splitGeneratorParallelism)
+    {
+        this.splitGeneratorParallelism = splitGeneratorParallelism;
+        return this;
+    }
+
+    public int getSplitGeneratorParallelism()
+    {
+        return this.splitGeneratorParallelism;
+    }
+
+    @Config("hudi.min-partition-batch-size")
+    public HudiConfig setMinPartitionBatchSize(int minPartitionBatchSize)
+    {
+        this.minPartitionBatchSize = minPartitionBatchSize;
+        return this;
+    }
+
+    @Min(1)
+    @Max(100)
+    public int getMinPartitionBatchSize()
+    {
+        return minPartitionBatchSize;
+    }
+
+    @Config("hudi.max-partition-batch-size")
+    public HudiConfig setMaxPartitionBatchSize(int maxPartitionBatchSize)
+    {
+        this.maxPartitionBatchSize = maxPartitionBatchSize;
+        return this;
+    }
+
+    @Min(1)
+    @Max(1000)
+    public int getMaxPartitionBatchSize()
+    {
+        return maxPartitionBatchSize;
+    }
+
+    @Config("hudi.size-based-split-weights-enabled")
+    @ConfigDescription("Unlike uniform splitting, size-based splitting ensures that each batch of splits has enough data to process. " +
+            "By default, it is enabled to improve performance")
+    public HudiConfig setSizeBasedSplitWeightsEnabled(boolean sizeBasedSplitWeightsEnabled)
+    {
+        this.sizeBasedSplitWeightsEnabled = sizeBasedSplitWeightsEnabled;
+        return this;
+    }
+
+    public boolean isSizeBasedSplitWeightsEnabled()
+    {
+        return sizeBasedSplitWeightsEnabled;
+    }
+
+    @Config("hudi.standard-split-weight-size")
+    @ConfigDescription("The split size corresponding to the standard weight (1.0) "
+            + "when size based split weights are enabled")
+    public HudiConfig setStandardSplitWeightSize(DataSize standardSplitWeightSize)
+    {
+        this.standardSplitWeightSize = standardSplitWeightSize;
+        return this;
+    }
+
+    @NotNull
+    public DataSize getStandardSplitWeightSize()
+    {
+        return standardSplitWeightSize;
+    }
+
+    @Config("hudi.minimum-assigned-split-weight")
+    @ConfigDescription("Minimum weight that a split can be assigned when size based split weights are enabled")
+    public HudiConfig setMinimumAssignedSplitWeight(double minimumAssignedSplitWeight)
+    {
+        this.minimumAssignedSplitWeight = minimumAssignedSplitWeight;
+        return this;
+    }
+
+    @DecimalMax("1")
+    @DecimalMin(value = "0", inclusive = false)
+    public double getMinimumAssignedSplitWeight()
+    {
+        return minimumAssignedSplitWeight;
+    }
+
+    @Min(1)
+    public int getMaxSplitsPerSecond()
+    {
+        return maxSplitsPerSecond;
+    }
+
+    @Config("hudi.max-splits-per-second")
```

Comment:
Please fill in a description for all the configs to better understand the functionality behind these settings.

Reply:
will do